openid-4-verifiable-presentations-1_0.md

```diff
@@ -542,11 +548,250 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+that claim is not present or does not match the restriction expressed by `value` or `values`. Any
+claim in a group listed in `alternatives` MUST NOT itself contain an `alternatives` property.
+
+### Semantics of `alternative_claims` and `alternative_credentials`
+
+By default, each element within `expect_credentials` or `expect_claims` MUST be returned, unless:
+
+- The element contains an `alternative_claims` or `alternative_credentials`
+  property and for at least one of the groups listed as an alternative, all
+  elements in the group can be returned.
+- The element is part of a group that is itself listed as an alternative in
+  another element and returning the element is not needed to fulfill the
+  previous rule.
+
+Groups of claims are scoped to the specific credential; i.e., groups with the
+same identifier that appear in separate Credential Query in `expect_credentials`
+have no relation to each other.
+
+## Format-specific Properties {#format_specific_properties}
+
+The following format-specific properties are defined:
+
+### Format `vc+sd-jwt` {#format_vc_sd_jwt}
+
+`vct_values`:
+: OPTIONAL. An array of strings that specifies allowed values for
+the type of the requested Verifiable Credential. All elements in the array MUST
+be valid type identifiers as defined in [@!I-D.ietf-oauth-sd-jwt-vc]. The Wallet
+MAY return credentials that inherit from any of the specified types, following
+the inheritance logic defined in [@!I-D.ietf-oauth-sd-jwt-vc].
+
+`credential_signing_alg_values`:
+: OPTIONAL. An array of strings that specifies
+the allowed algorithms for signing the SD-JWT (i.e., only the issuer-signed part).
+
+`kbjwt_signing_alg_values`:
+: OPTIONAL. An array of strings that specifies the
+allowed algorithms for signing the Key-Binding JWT (KB-JWT) (i.e., the part signed
+by the holder).
+
+`sd_alg_values`:
+: OPTIONAL. An array of strings that specifies the allowed hash algorithms for
+digests in the SD-JWT and KB-JWT.
+
+### Format `mso_mdoc` {#format_mso_mdoc}
+
+TBD
+
+### Format `jwt_vp*`
+
+TBD
+
+
+## Claims Path Pointer {#claims_path_pointer}
+
+A claims path pointer is a pointer into the JSON structure of the Verifiable
+Credential, identifying one or more claims. A claims path pointer MUST be a
+non-empty array of strings and non-negative integers. A `*` string value indicates
+that all elements of the currently selected array(s) are to be selected; any other
+string indicates that the respective key is to be selected; and a non-negative
+integer indicates that the respective index in an array is to be selected. The path
+is formed as follows:
+
+  - Start with an empty array.
+  - To address a particular claim within an object, append the key (claim name)
+    to the array.
+  - To address an element within an array, append the index to the array (as a
+    non-negative, 0-based integer).
+  - To address all elements within an array, append the string `*` to the array.
```

Comment:
With PEX it's possible to query for a specific item in an array based on value. Would that be possible?

I.e. i need to prove my german nationality, but my other nationalities are not important to disclose. This is something SD-JWT disclosures can support, but may be complex to convey in a vp query

Reply:
Please see a proposal here: https://github.com/openid/OpenID4VP/pull/266#issuecomment-2362093202